src/Ormolu/Logging.hs

```diff
@@ -0,0 +1,84 @@
+module Ormolu.Logging
+  ( initializeLogging,
+    logDebug,
+    logDebugM,
+    logWarn,
+    logWarnM,
+    logError,
+    logErrorM,
+  )
+where
+
+import Data.Foldable (traverse_)
+import Data.IORef (IORef, newIORef, readIORef, writeIORef)
+import Ormolu.Config (Config (..))
+import System.IO (hPutStrLn, stderr)
+import System.IO.Unsafe (unsafePerformIO)
+
+data LoggerConfig = LoggerConfig
+  { debugEnabled :: Bool
+  }
+
+loggerConfig :: IORef LoggerConfig
+loggerConfig = unsafePerformIO $ newIORef (error "Logger not configured yet")
+{-# NOINLINE loggerConfig #-}
+
+initializeLogging :: Config region -> IO ()
+initializeLogging cfg =
+  writeIORef loggerConfig $
+    LoggerConfig
+      { debugEnabled = cfgDebug cfg
+      }
```

Comment:
Yeah, this is a bit hacky and unprincipled, smuggling the config through the global environment. But I figured this was much neater than storing the config in `R`'s environment and passing it through `addFixityInfo`.

I call this function immediately in all the entrypoints (`main`, `ormolu`, and for tests), so we should be good in principle. The only issue might be if someone imports from Ormolu's submodules + calls functions directly. I guess I could set a default logger config instead of erroring with "logger not configured yet".